src/mame/jaleco/megasys1.cpp

```diff
@@ -5139,33 +5113,91 @@ void megasys1_state::megasys1A_mcu_hs_w(offs_t offset, u16 data, u16 mem_mask)
 		printf("MCU HS W %04x (%04x) -> [%02x]\n",data,mem_mask,offset*2);
 }
 
-void megasys1_state::init_astyanax()
+void megasys1_state::install_gatearray_overlay(uint32_t base_write, const uint16_t* sequence)
 {
-	astyanax_rom_decode(machine(), "maincpu");
-	m_maincpu->space(AS_PROGRAM).install_read_handler(0x00000, 0x3ffff, read16s_delegate(*this, FUNC(megasys1_state::megasys1A_mcu_hs_r)));
-	m_maincpu->space(AS_PROGRAM).install_write_handler(0x20000, 0x20009, write16s_delegate(*this, FUNC(megasys1_state::megasys1A_mcu_hs_w)));
-
+	m_maincpu->space(AS_PROGRAM).install_read_handler(0x00000, 0x3ffff, read16s_delegate(*this, FUNC(megasys1_state::gatearray_r)));
+	m_maincpu->space(AS_PROGRAM).install_write_handler(base_write, base_write+0x9, write16s_delegate(*this, FUNC(megasys1_state::gatearray_w)));
+	m_gatearray_seq = sequence;
 	m_mcu_hs = 0;
 	memset(m_mcu_hs_ram, 0, sizeof(m_mcu_hs_ram));
-
 	save_item(NAME(m_mcu_hs));
 	save_item(NAME(m_mcu_hs_ram));
 }
```

Comment:
Please put the right-associative unary `*` qualifier to the right of the space, and restore the blank lines that separated installing memory handlers, intialising members, and registering items for save states.